data/src/test/java/org/apache/iceberg/TestScanPlanningAndReporting.java

```diff
@@ -0,0 +1,334 @@
+    assertThat(scanReport.scanMetrics().scannedDataManifests().value()).isEqualTo(1);
+    assertThat(scanReport.scanMetrics().skippedDataManifests().value()).isEqualTo(0);
+    assertThat(scanReport.scanMetrics().totalDataManifests().value()).isEqualTo(1);
+    assertThat(scanReport.scanMetrics().totalDeleteManifests().value()).isEqualTo(0);
+    assertThat(scanReport.scanMetrics().totalFileSizeInBytes().value()).isGreaterThan(0L);
+    assertThat(scanReport.scanMetrics().totalDeleteFileSizeInBytes().value()).isEqualTo(0L);
+
+    // all files
+    try (CloseableIterable<FileScanTask> fileScanTasks =
+        tableScan.filter(Expressions.lessThan("x", "52")).planFiles()) {
+      fileScanTasks.forEach(task -> {});
+    }
+
+    scanReport = reporter.lastReport();
+    assertThat(scanReport).isNotNull();
+    assertThat(scanReport.tableName()).isEqualTo(tableName);
+    assertThat(scanReport.snapshotId()).isEqualTo(1L);
+    assertThat(scanReport.scanMetrics().totalPlanningDuration().totalDuration())
+        .isGreaterThan(Duration.ZERO);
+    assertThat(scanReport.scanMetrics().resultDataFiles().value()).isEqualTo(3);
+    assertThat(scanReport.scanMetrics().resultDeleteFiles().value()).isEqualTo(0);
+    assertThat(scanReport.scanMetrics().scannedDataManifests().value()).isEqualTo(1);
+    assertThat(scanReport.scanMetrics().skippedDataManifests().value()).isEqualTo(0);
+    assertThat(scanReport.scanMetrics().totalDataManifests().value()).isEqualTo(1);
+    assertThat(scanReport.scanMetrics().totalDeleteManifests().value()).isEqualTo(0);
+    assertThat(scanReport.scanMetrics().totalFileSizeInBytes().value()).isGreaterThan(0L);
+    assertThat(scanReport.scanMetrics().totalDeleteFileSizeInBytes().value()).isEqualTo(0L);
+  }
+
+  private Table createTableWithCustomRecords(String tableName) throws IOException {
+    Schema schema =
+        new Schema(
+            required(1, "id", Types.IntegerType.get()), required(2, "x", Types.StringType.get()));
+
+    Table tbl =
+        TestTables.create(
+            tableDir,
+            tableName,
+            schema,
+            PartitionSpec.builderFor(schema).build(),
+            SortOrder.unsorted(),
+            formatVersion,
+            reporter);
+    GenericRecord record = GenericRecord.create(schema);
+    record.setField("id", 1);
+    record.setField("x", "23");
+    GenericRecord record2 = GenericRecord.create(schema);
+    record2.setField("id", 2);
+    record2.setField("x", "30");
+    GenericRecord record3 = GenericRecord.create(schema);
+    record3.setField("id", 3);
+    record3.setField("x", "45");
+    GenericRecord record4 = GenericRecord.create(schema);
+    record4.setField("id", 3);
+    record4.setField("x", "51");
+    DataFile dataFile = writeParquetFile(tbl, Arrays.asList(record, record3));
+    DataFile dataFile2 = writeParquetFile(tbl, Arrays.asList(record2));
+    DataFile dataFile3 = writeParquetFile(tbl, Arrays.asList(record4));
```

Comment:
Can you add some whitespace here to make this more readable?

Maybe break up record creation from writes. Also, if you use Iceberg's `GenericRecord`, it's usually easier to create records. You just create one from the schema and call `copy(Map<String, Object>)` to make a new one with data values.

Reply:
thanks for the tip. updated. I left record creation and writes together for now, but I'm planning to re-visit and improving it later (when adding & testing new metrics)